drivers/console/efi_console.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief EFI console driver
+ *
+ * @details EFI console driver.
+ * Hooks into the printk and fputc (for printf) modules.
+ */
+
+#include <stdio.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+
+
+extern int efi_console_putchar(int c);
+
+#if defined(CONFIG_PRINTK) || defined(CONFIG_STDOUT_CONSOLE)
```

Comment:
I would suggest using
```cpp
if (IS_ENABLED(CONFIG_PRINTK) || defined(CONFIG_STDOUT_CONSOLE)
```
not 100% sure if this should be necessary though - is the conditional compile for `CONFIG_X86_EFI_CONSOLE` sufficient?

Reply:
Same as the top comment about IS_ENABLED macro. Thanks for your review.